unstructured/partition/__init__.py

```diff
@@ -40,6 +39,15 @@ def _partition_via_api(
 
     if response.status_code == 200:
         pages = response.json()["pages"]
-        return [element for page in pages for element in page["elements"]]
+        num_pages = len(pages)
+        elements = list()
+        for i, page in enumerate(pages):
+            for element in page["elements"]:
+                elements.append(element)
+            if include_page_breaks and i < num_pages - 1:
+                elements.append({"type": "PageBreak"})
```

Comment:
Just curious, why'd you decide to handle this here rather than at the endpoint?

Reply:
Main reason is that the endpoint returns representation of the document layout, so you already have `document["pages"]`. You don't really need to do anything additional to track pages there since it's already doing it.